src/App.tsx

```diff
@@ -1,14 +1,75 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import * as todosAPI from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isloading, setIsLoading] = useState(true);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filterStatus, setFilterStatus] = useState<string>('all');
+  const [query, setQuery] = useState<string>('');
+
+  const handleFilterChange = (status: string) => {
+    setFilterStatus(status);
+  };
+
+  const handleSelectedTodo = (todo: Todo) => {
+    setSelectedTodo(todo);
+  };
+
+  const handleCloseTodo = () => {
+    setSelectedTodo(null);
+  };
+
+  const handleQueryChange = (value: string) => {
+    setQuery(value);
+  };
+
+  const filteredTodos = todos
+    .filter((todo) => {
+      if (filterStatus === 'all') {
+        return true;
+      }
+
+      if (filterStatus === 'active') {
+        return !todo.completed;
+      }
+
+      if (filterStatus === 'completed') {
+        return todo.completed;
+      }
+
+      return true;
+    })
+    .filter((todo) => {
```

Comment:
please consider moving this outside of component, as utility function - we want to keep our components as lean as possible

Reply:
Moved above the app component.